client/src/js/services/CashService.js

```diff
@@ -153,33 +158,62 @@ function CashService(Api, Exchange, Session, moment) {
       .replace(':user', Session.user.display_name);
   }
 
+
   /**
    * @method formatFilterParameters
    * @description format filters parameters
    */
- function formatFilterParameters(params) {
-   var columns = [
-     { field: 'is_caution', displayName: 'FORM.LABELS.CAUTION' },
-     { field: 'cashbox_id', displayName: 'FORM.LABELS.CASHBOX' },
-     { field: 'debtor_uuid', displayName: 'FORM.LABELS.CLIENT' },
-     { field: 'user_id', displayName: 'FORM.LABELS.USER' },
-     { field: 'reference', displayName: 'FORM.LABELS.REFERENCE' },
-     { field: 'dateFrom', displayName: 'FORM.LABELS.DATE_FROM', comparitor: '>', ngFilter:'date' },
-     { field: 'dateTo', displayName: 'FORM.LABELS.DATE_TO', comparitor: '<', ngFilter:'date' },
-   ];
-
-   // returns columns from filters
-   return columns.filter(function (column) {
-     var value = params[column.field];
-
-     if (angular.isDefined(value)) {
-       column.value = value;
-       return true;
-     } else {
-       return false;
-     }
-   });
- }
+  function formatFilterParameters(params) {
+    var columns = [
+      { field: 'is_caution', displayName: 'FORM.LABELS.CAUTION' },
+      { field: 'cashbox_id', displayName: 'FORM.LABELS.CASHBOX' },
+      { field: 'debtor_uuid', displayName: 'FORM.LABELS.CLIENT' },
+      { field: 'user_id', displayName: 'FORM.LABELS.USER' },
+      { field: 'reference', displayName: 'FORM.LABELS.REFERENCE' },
+      { field: 'dateFrom', displayName: 'FORM.LABELS.DATE_FROM', comparitor: '>', ngFilter:'date' },
+      { field: 'dateTo', displayName: 'FORM.LABELS.DATE_TO', comparitor: '<', ngFilter:'date' },
+    ];
+
+    // returns columns from filters
+    return columns.filter(function (column) {
+      var value = params[column.field];
+
+      if (angular.isDefined(value)) {
+        column.value = value;
+        return true;
+      } else {
+        return false;
+      }
+    });
+  }
+
+  //open a dialog box to Cancel Cash Paiement
+  function openCancelCashModal(invoice) {
+    return Modal.open({
+      templateUrl : 'partials/cash/modals/modalCancelCash.html',
+      resolve : { data : { invoice : invoice } },
+      size : 'md',
+      animation : true,
+      keyboard  : false,
+      backdrop : 'static',
+      controller : 'ModalCancelCashController as ModalCtrl',
+    }).result;
+  }
+
+  /**
+   * @desc It checkCashPayment the invoice from the database
+   * @param {String} invoiceUuid, is the uuid of invoice
+   * @example
+   * service.checkCashPayment(invoiceUuid)
+   * .then(function (res){
+   *   your code here
+   *  });
+   */
+  function checkCashPayment (invoiceUuid){
+    var url = urlCheckin + invoiceUuid;
+    return $http.get(url)
```

Comment:
For future reference, you can get both `$http` and `util` from the `PrototypeApiService`.  Like this:
```js
var service = Api('/some/url/');

function checkCashPayment(invoiceUuid) {
  var url = service.url + invoiceUuid;
  return service.$http.get(url)
   .then(service.util.unwrapHttpResponse);
}
```